### DISCUSSION TITLE
port kill

### DISCUSSION THREAD

**Discussion by youngkyo0504** [johnlindquist/kit#1038] [Share]
[Open Port kill in Script Kit](https://scriptkit.com/api/new?name=time-from-now&url=https://gist.githubusercontent.com/youngkyo0504/f8632d4775b7964188dc714538c81991/raw/b38e4402c31c320d427077feff73aba65f60a190/port-kill.js
)

|Success|Fail|
|---|---|
|<img width="500" alt="image" src="https://user-images.githubusercontent.com/78121870/213914239-c9695eda-9b65-4918-9b9b-31d464c35a1d.png">|<img width="500" alt="image" src="https://user-images.githubusercontent.com/78121870/213914154-ddab5451-c486-4c0c-9adc-5d98a34efbc5.png">|

```js
// Name: Port kill
// Description: Enter port number to kill process listening on port.
// Author: kyo young
// GitHub: @youngkyo0504

import '@johnlindquist/kit';

const killPort = await npm('kill-port');
const port = await arg('Enter port to kill')
const containerClassName = 'flex justify-center items-center text-4xl h-full'

try {
    await killPort(port, 'tcp')
    await div(`🤖 listening on port ${port} has been killed.`,containerClassName);
} catch (error) {
    console.error(error);
    await div(`
    🛰️ ${error.message}
    `,containerClassName)
}
```

**Comment by mabry1985** [comment]
this is great, thx
